backend/repository/endpoint_update.go

```diff
@@ -161,7 +161,7 @@ func (r *Repository) deleteEndpointConfig(endpointCfgDir string, endpointName st
 }
 
 func getEndpointsWithMiddleware(gatewayCfg *Config, middleware string, endpointGroup string) []string {
-	endpoints := make([]string, 0)
+	var endpoints []string
```

Comment:
Why is this better? This creates a slice of size 0, that will exponentially grow as we append. Can we not do something like

```
endpoints := make([]string, 0, len(gatewayCfg.Endpoints))
```

Reply:
It simply passes the linter, no idea how it is on master and passing linters.

Reply:
I'm not sure about this repo but it is possible in Github to enforce CI passing before being allowed to land on master using the Github UI. Perhaps we should turn that on to nudge people to pass CI including lint?

Reply:
The repo has pre-push git hooks that enforces all checks to pass before pushing to remote, which I feel somewhat enough. But still I don't know how things like that was on master at the first place.